### DISCUSSION TITLE
Not able to run nextcloud-aio on a QNAP

### DISCUSSION THREAD

**Discussion by HolgerDoerrschuck** [nextcloud/all-in-one#4172] [Questions] [answered]
Hi, I am not able to run nextcloud-aio on a QNAP container. After starting the container it stopps immediatly. The log shows the following message: 

> Cannot connect to the docker socket. Cannot proceed.
> Did you maybe remove group read permissions for the docker socket? AIO needs them in order to access the docker socket.
> If SELinux is enabled on your host, see https://github.com/nextcloud/all-in-one#are-there-known-problems-when-selinux-is-enabled
> If you are on TrueNas SCALE, see https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale

I tried it on two different QNAP Systems:  TS-253D and a new TS-264. Both end in the same error. 

I use the container in the following way: I use bridge mode to define a separate IP addrress not to run into issues with ports on used by QNAP QTS. I set the environment variable (as documented for Synology) and the storage with:

/Container/container-station-data/lib/docker/var/run/docker.sock :/var/run/docker.sock
/Container/nextcloud_aio :/mnt/docker-aio-config

see all screenshots attached.
[screenshots.pdf](https://github.com/nextcloud/all-in-one/files/14155847/screenshots.pdf)

Because  of despair I set all rights for all user and groups on the new one - the command show:

> ls -l /var/run/docker.sock
> srw-rw---- 1 admin administrators 0 2024-02-03 17:03 /var/run/docker.sock=

Please help - I have no idea what else I can do!

Thanks Holger

**Comment by szaimen** [comment]
Hi, can you post the complete docker run command or the docker compose file here that you used?

**Reply by HolgerDoerrschuck** [reply]
Hi, I have no run command or compose file. I did it via QTS. I attached all screenshots to that post. I just added in the environment: WATCHTOWER_DOCKER_SOCKET_PATH :/Container/container-station-data/lib/docker/var/run/docker.sock
and the two storages mentioned above

**Reply by szaimen** [reply, marked as answer]
Ah I see. Are you sure this path is actually correct? `/Container/container-station-data/lib/docker/var/run/docker.sock`
Also what is the output of `sudo ls -l /Container/container-station-data/lib/docker/var/run/docker.sock` 
And is SELinux enabled?

**Reply by HolgerDoerrschuck** [reply]
`sudo ls -l /Container/container-station-data/lib/docker/var/run/docker.sock` shows:
`ls: cannot access /Container/container-station-data/lib/docker/var/run/docker.sock: No such file or directory`

runing "`docker info`" shows:
`Docker Root Dir: /share/CACHEDEV1_DATA/Container/container-station-data/lib/docker`

I tried again with the path: `/Container/container-station-data/lib/docker/docker.sock` 

this results again into the error
```
Cannot connect to the docker socket. Cannot proceed.
Did you maybe remove group read permissions for the docker socket? AIO needs them in order to access the docker socket.
If SELinux is enabled on your host, see https://github.com/nextcloud/all-in-one#are-there-known-problems-when-selinux-is-enabled
If you are on TrueNas SCALE, see https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale

```
SELinux? I don't know that

**Reply by szaimen** [reply]
I think you have still not found out where the docker socket actually is located. Probably makes sense to google this and then confirm with `sudo ls -l`...

**Reply by HolgerDoerrschuck** [reply]
Hi all, a long time is passed but now I make progress. You can find my YAML code at the end of the post. To adapt it for your QNAP Ccntainer you have to define the IP (bridge mode) and network name: qnet-static-eth1-4f3222 

Everything looks fine. The setup starts after asking the passphrase but now I get stucked. When I enter the url https://192.168.68.111:8080 it jumps to the https://192.168.68.111:8080/containers and shows the screen attached
![Screenshot 2024-08-29 072437](https://github.com/user-attachments/assets/b23c750a-ac03-40cf-a68d-ef07f656cbc6)

I get more and more frustrated with my QNAP. Maybe I should sell it and buy a Synology. Any idea? Do I still have installation issues?

Thanks Holger


`version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer # Diese Zeile darf nicht geändert werden, da sonst die integrierte Backup-Lösung nicht funktioniert

services:
  nextcloud:
    dns:
      - 8.8.8.8   # Google DNS
      - 1.1.1.1   # Cloudflare DNS  
    image: nextcloud/all-in-one:latest
    restart: always
    container_name: nextcloud-aio-mastercontainer # Diese Zeile darf nicht geändert werden, da sonst AIO nicht korrekt funktioniert
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config # Diese Zeile darf nicht geändert werden, da sonst die integrierte Backup-Lösung nicht funktioniert
      - /var/run/docker.sock:/var/run/docker.sock:ro # Kann auf macOS, Windows oder Docker Rootless geändert werden. Siehe die entsprechende Dokumentation. Bei Anpassungen nicht vergessen, auch 'WATCHTOWER_DOCKER_SOCKET_PATH' zu setzen!
    networks:
      nextcloud_network:
        ipv4_address: 192.168.68.111
    environment: # Wird benötigt, wenn eine der untenstehenden Optionen verwendet wird
      - AIO_DISABLE_BACKUP_SECTION=true # Setting this to true allows to hide the backup section in the AIO interface. See https://github.com/nextcloud/all-in-one#how-to-disable-the-backup-section
      - APACHE_PORT=80 # Verwendung der Standardports
      - APACHE_IP_BINDING=127.0.0.1 # Sollte gesetzt werden, wenn hinter einem Webserver oder Reverse Proxy betrieben wird, der auf dem gleichen Host läuft. Siehe https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      - NEXTCLOUD_DATADIR=/mnt/ncdata # Erlaubt das Setzen des Host-Verzeichnisses für Nextclouds Datadir. Siehe https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
      - NEXTCLOUD_MOUNT=/mnt/ # Allows the Nextcloud container to access the chosen directory on the host. See https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host 
      - NEXTCLOUD_UPLOAD_LIMIT=10G # Kann angepasst werden, falls du mehr benötigst. Siehe https://github.com/nextcloud/all-in-one#how-to-adjust-the-upload-limit-for-nextcloud
      - NEXTCLOUD_MAX_TIME=3600 # Kann angepasst werden, falls du mehr benötigst. Siehe https://github.com/nextcloud/all-in-one#how-to-adjust-the-max-execution-time-for-nextcloud
      - NEXTCLOUD_MEMORY_LIMIT=512M # Kann angepasst werden, falls du mehr benötigst. Siehe https://github.com/nextcloud/all-in-one#how-to-adjust-the-php-memory-limit-for-nextcloud
      - NEXTCLOUD_ADDITIONAL_APKS=imagemagick # Dies erlaubt es, zusätzliche Pakete dauerhaft zum Nextcloud-Container hinzuzufügen. Standard ist imagemagick, kann jedoch durch das Modifizieren dieses Werts überschrieben werden. Siehe https://github.com/nextcloud/all-in-one#how-to-add-os-packages-permanently-to-the-nextcloud-container
      - WATCHTOWER_DOCKER_SOCKET_PATH=/var/run/docker.sock # Needs to be specified if the docker socket on the host is not located in the default '/var/run/docker.sock'. Otherwise mastercontainer updates will fail. For macos it needs to be '/var/run/docker.sock'
 
networks:
  nextcloud_network:
    external: true
    name: qnet-static-eth1-4f3222 
    driver: bridge
    ipam:
      config:
        - subnet: 192.168.68.0/24
`

**Reply by HolgerDoerrschuck** [reply]
the docker log also looks good 

![Screenshot 2024-08-29 073132](https://github.com/user-attachments/assets/d520385d-af10-42c4-957e-3872156375a1)

**Reply by navidzol** [reply]
Do you see all the dockers starting up?

**Comment by purplekairi1312** [comment]
Im having this same issue
Does someone know please where  /var/run/docker.sock is located commonly on a QNAP with its QNAP OS?

**Reply by lhsei** [reply]
Hello,
I am using nextcloud-aio on my TS-253D without problems. I have described the way at 
https://forum.qnapclub.de/thread/65097-nextcloud-mittels-docker-installieren/?postID=523481#post523481
